qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -350,6 +369,9 @@ def _score_heuristic(self, heuristic, front_layer, extended_set, layout, swap_qu
         first_cost = self._compute_cost(front_layer, layout)
         if heuristic == "basic":
             return first_cost
+        elif heuristic == "lookahead":
+            first_cost = self._score_heuristic("basic", front_layer, [], layout)
+            first_cost /= len(front_layer)
 
         first_cost /= len(front_layer)
         second_cost = 0
```

Comment:
Naively, this looks inefficient if `heuristic == "lookahead"` because it throws away the initial cost calculation.  It also appears to divide by the length of the front layer twice now.